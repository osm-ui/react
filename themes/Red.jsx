
import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import _ from 'lodash';
import colors from '../colors';
import { buildDarkThemeConfig } from '../helpers/themes';
import {
    config as defaultThemeConfig,
    StyledSpan,
} from './Default';


export const config = _.merge(
    {},
    buildDarkThemeConfig(defaultThemeConfig, colors, 'red'),
    {
        form: {
            button: {
                danger: {
                    color: colors.white,
                    backgroundColor: colors.red1,
                    borderColor: colors.red1,
                    hoverBackgroundColor: colors.red3,
                    hoverBorderColor: colors.red3,
                    focusBackgroundColor: colors.red3,
                    focusBorderColor: colors.red3,
                    activeBackgroundColor: colors.red3,
                    activeBorderColor: colors.red3,
                },
            },
        },
    },
);


const RedTheme = ({ children, ...rest }) => (
    <ThemeProvider theme={config}>
        <StyledSpan>
            {
                React.Children.map(children,
                    (child) => {
                        if (React.isValidElement(child)) {
                            return React.cloneElement(child, rest);
                        }

                        return child;
                    }
                )
            }
        </StyledSpan>
    </ThemeProvider>
);


RedTheme.propTypes = {
    children: PropTypes.node.isRequired,
};

RedTheme.displayName = 'RedTheme';

export default RedTheme;
