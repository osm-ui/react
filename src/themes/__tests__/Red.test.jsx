import React from 'react';
import {
  snapshotWithElementChildren,
  snapshotWithTextChildren
} from 'helpers/tests';
import Red from 'themes/Red';

describe('When using snapshots', () => {
  it('Should render with an element children', () =>
    snapshotWithElementChildren(Red));

  it('Should render with a text children', () => snapshotWithTextChildren(Red));
});
