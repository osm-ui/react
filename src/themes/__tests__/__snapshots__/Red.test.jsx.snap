// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When using snapshots Should render with a text children 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif;
}

<span
  className="c0"
>
  <span
    className="c0"
  >
    This is a text
  </span>
</span>
`;

exports[`When using snapshots Should render with an element children 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif;
}

<span
  className="c0"
>
  <span
    className="c0"
  >
    <div />
  </span>
</span>
`;
